src/MessageInput.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+
+const api = require('./api.js');
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+  textField: {
+    width: '30%',
+    marginRight: 15
+  }
+});
+
+class MessageInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      id: `${this.props.id}`,
+      index: `${this.props.index}`,
+      prob: `${this.props.prob}`,
+      text: `${this.props.text}`,
+      success: false,
+      error: false,
+      warning: false
+    }
+  }
+
+  handleOnChange = event => {
+    console.log('Click');
+    console.log(event.target.value);
+  };
+
+  updateText = event => {
+    this.setState({
+      text: `${event.target.value}`
+    })
+  }
+
+  updateProb = event => {
+    this.setState({
+      prob: `${event.target.value}`
+    })
+  }
+
+  open = (result) => {
+    console.log(result)
+    if (result.errors) {
+      if (this.props.pass === "") {
+        this.setState({warning: true})
+      } else {
+        this.setState({error: true})
+      }
+    } else {
+      this.setState({success: true})
+    }
+  }
+
+  close = (event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    if (this.state.error) {
+      this.setState({error: false})
+    } else if (this.state.warning){
+      this.setState({warning: false})
+    } else {
+      this.setState({success: false})
+    }
+  }
+
+  updateMessage = event => {
+    api.post(`glyphs/${this.state.id}/messages/${this.state.index}`,
+      {"pass" : this.props.pass, "text" : this.state.text, "probability" : this.state.prob}
+    ).then(result => {this.open(result)})
+  }
+
+  render() {
+    const { classes } = this.props;
+
+
+    return (
+      <div>
+        <form className={classes.container} noValidate autoComplete="off">
+          <TextField
+            id="outlined-basic"
+            label={`Message ${this.props.index}`}
+            defaultValue={this.props.text}
+            className={classes.textField}
+            margin="normal"
+            variant="outlined"
+            size="small"
+            onChange={this.updateText}
+            onBlur={this.updateMessage}
+            multiline
+          />
+          <TextField
+            type="number"
+            style={{width: 75}}
+            id="outlined-basic"
+            label={`Prob ${this.props.index}`}
+            variant="outlined"
+            size="small"
+            margin="normal"
+            defaultValue={this.props.prob}
+            onChange={this.updateProb}
+            onBlur={this.updateMessage}
+          />
+        </form>
+        <Snackbar open={this.state.success} autoHideDuration={3000} onClose={this.close}>
```

Comment:
I feel like this display should probably be attached to the `MessageEditor` component, and it can simply see if any of its children have messages to display at all times.  Just an idea, though.